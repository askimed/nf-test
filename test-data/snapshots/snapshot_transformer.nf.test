nextflow_function {

  name "Test Function Say Hello"

  script "./functions.nf"
  function "create_object"

  test("Passing case") {

    when {
      function {
        """
        input[0] = "aaron"
        """
      }
    }

    then {
      assert snapshot(function.result).
          view().
          view(true).
          remove('.start_time').
          replace('.end_time', "<END_TIME>").
          map('.chunks', chunks -> chunks.sort()).

          // vs. value is each element. no the list itself.
          // map('.chunks[*]', value -> 27).

          map('contents[0].value', value -> format("#0.00", value)).
          //remove('contents[0].files[?(@ == "output.log")]').
          //remove('contents[0].files[?(@ =~ /.*\\.log$/)]').

          //traverse( (key, value) ->  {
          //  value == "output.log" ? "<LOG>" : value
          //}).
          //traverse( (key, value) ->  {
          //  value.toString().endsWith(".log") ? "<LOG>" : value
          //}).
          traverse( (key, value) ->  {
            return value.toString().endsWith(".log") ? file(value).name : value
          }).
          map('contents[0].files[0]', value -> file(value).readLines()).

          view().
          match()
    }

  }

}