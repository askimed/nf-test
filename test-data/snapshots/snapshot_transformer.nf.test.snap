{
    "Passing case": {
        "content": [
            {
                "id": "1234-5678-9101",
                "status": "SUCCESS",
                "end_time": "<END_TIME>",
                "chunks": [
                    1,
                    2,
                    3,
                    4,
                    5,
                    6,
                    7,
                    8,
                    9
                ],
                "value": "1.36",
                "files": [
                    [
                        "ID,Name,Quantity,Price",
                        "1,Item1,11,2.49",
                        "2,Item2,12,2.99",
                        "3,Item3,13,3.49",
                        "4,Item4,14,3.99",
                        "5,Item5,15,4.49",
                        "6,Item6,16,4.99",
                        "7,Item7,17,5.49",
                        "8,Item8,18,5.99",
                        "9,Item9,19,6.49",
                        "10,Item10,20,6.99",
                        "11,Item11,21,7.49",
                        "12,Item12,22,7.99",
                        "13,Item13,23,8.49",
                        "14,Item14,24,8.99",
                        "15,Item15,25,9.49",
                        "16,Item16,26,9.99",
                        "17,Item17,27,10.49",
                        "18,Item18,28,10.99",
                        "19,Item19,29,11.49",
                        "20,Item20,30,11.99"
                    ],
                    "output.log"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0-rc2",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-06-12T17:08:37.459381"
    }
}